Snooper/src/main/java/com/prateekj/snooper/AndroidSnooper.java

```diff
@@ -74,6 +76,10 @@ public static AndroidSnooper init(Application application) {
     androidSnooper.context = application;
     androidSnooper.snooperRepo = new SnooperRepo(androidSnooper.context);
     androidSnooper.shakeDetector = new ShakeDetector(new SnooperShakeListener(androidSnooper));
+    androidSnooper.writeThread = new HandlerThread("AndroidSnooper:Writer");
+    androidSnooper.writeThread.start();
+    androidSnooper.writeHandler = new Handler(androidSnooper.writeThread.getLooper());
+
```

Comment:
HI @nullptr2this , I think we should call `androidSnooper.writeThread.quit()` to free the resources. Maybe when the app is being killed. I am not sure of the performance implications of `HandlerThread` as it will keep running for a long time.

Reply:
When the application is killed the resource will be reclaimed anyway.  Process death will result in thread death.  There are no OS or inter-process resources being held so the thread will not retain any other resources when it is killed by the process being killed.

HandlerThread is a wrapper around a looper.  It's essentially a single thread executor that also accepts Messages in addition to runnables.  Like the main thread it will not execute any code if it has no messages or runnables in it's queue.

A natural extension to this is to provide the AndroidSnooper client the ability to provide their own executor during initialization, but as:

1) The existing code was already posting a runnable to a handler (on the main thread).
2) I was uncertain as to how receptive you would be to PRs. :-)

I elected to keep the mechanism handler based for this PR.